### PR TITLE
[SYCL][Test] Add negative checks for the variadic O0 printf LIT

### PR DESCRIPTION
**Pull request by AGindinson**
This is a follow-up to https://github.com/intel/llvm/pull/5286.

In addition, reflect the default PM switch in the existing `RUN` lines.

Signed-off-by: Artem Gindinson <artem.gindinson@intel.com>

### CODE REVIEW

**Comment by kbobrovs** [comment]
I think dpcpp-esimd-reviewers is assigned due to mistake in the code owners file, which is being fixed. But I will give my approval to enable merge, hoping others will do actual review.

**Review by kbobrovs — APPROVED** [review]

**Comment by AGindinson** [comment]
> I think dpcpp-esimd-reviewers is assigned due to mistake in the code owners file, which is being fixed. But I will give my approval to enable merge, hoping others will do actual review.

Thanks @kbobrovs - this truly is by code-ownership-related mistake, and https://github.com/intel/llvm/pull/5287 should have resolved it now (thanks to your approval there, too!).

File: llvm/test/SYCLLowerIR/printf_addrspace/generic_as_negative_checks.ll
Comment (mlychkov) [inline llvm/test/SYCLLowerIR/printf_addrspace/generic_as_negative_checks.ll:15]:
```suggestion
; RUN: opt < %S/generic_as_variadic_no_opt.ll --SYCLMutatePrintfAddrspace -S --enable-new-pm=0 | FileCheck %s --check-prefixes=CHECK-WRAPPER,CHECK-BUILTIN
; RUN: opt < %S/generic_as_variadic_no_opt.ll -passes=SYCLMutatePrintfAddrspace -S | FileCheck %s --check-prefixes=CHECK-WRAPPER,CHECK-BUILTIN
```
This does the same checks for legacy and new PMs. However, when new PM becomes default suggested options will keep checks for both PMs, while current options will check only for new PM twice.
I'd suggest to change PM selection options for all other checks as well.

Comment (mlychkov) [reply llvm/test/SYCLLowerIR/printf_addrspace/generic_as_negative_checks.ll:15]: I think either option is fine. So, it's up to you.

Comment (AGindinson) [reply llvm/test/SYCLLowerIR/printf_addrspace/generic_as_negative_checks.ll]: https://github.com/intel/llvm/pull/5302/commits/bb9b948b10d55bdc680cae13d9f55d8dd945e69d for the new checks, https://github.com/intel/llvm/pull/5302/commits/74d315364d4cc723425a0b29a301b80cfbebd8bd for the existing ones.

**Review by AlexeySachkov — APPROVED** [review]

**Review by mlychkov — APPROVED** [review]

**Pull request merged**
